### PR TITLE
Fix URL redirects (e.g., to alpha/beta)

### PR DESCRIPTION
**Pull request by Zlatkovsky**

### CODE REVIEW

File: config/env.config.js [hunk 1]
```diff
@@ -85,15 +91,38 @@ class RedirectPlugin {
...
                                 return null;
-                            };
+                            }
+
+                            function isAllowedUrl(url) {
+                                if (url == null || url.trim().length === 0) {
+                                    return true;
+                                }
+                                
+                                url = decodeURIComponent(url);
```
Comment (dsaunders45459) [inline config/env.config.js]: Do we need to decodeURIComponent here, can't the caller be responsible for sending in a decoded URL?

File: config/env.config.js [hunk 2]
```diff
@@ -85,15 +91,38 @@ class RedirectPlugin {
...
+
+                                var validRedirectLocations = ${JSON.stringify(validRedirectLocations)};
```
Comment (dsaunders45459) [inline config/env.config.js]: validRedirectLocations is already defined as const

File: config/env.config.js [hunk 3]
```diff
@@ -85,15 +91,38 @@ class RedirectPlugin {
...
+
+                                var matchingLocation = validRedirectLocations.find(function(location) {
+                                    return location.indexOf(url) === 0;
```
Comment (dsaunders45459) [inline config/env.config.js]: We should probably do an exact match, not just looking for a substring as someone could register a new domain.

File: config/env.config.js [hunk 4]
```diff
@@ -85,15 +91,38 @@ class RedirectPlugin {
...
+                                });
+
+                                return matchingLocation ? true : false;
```
Comment (dsaunders45459) [inline config/env.config.js]: !!matchingLocation

Comment (dsaunders45459) [reply config/env.config.js]: actually this will treat the 0 index as false, we need to test for null

File: config/env.config.js [hunk 5]
```diff
@@ -126,9 +155,14 @@ class RedirectPlugin {
                             }
 
                             // If reached here, environment is already configured
-                        })();
+                            return;
+
+                        } catch (e) {
+                            console.log("Error redirecting the environments, staying on current page", e);
```
Comment (dsaunders45459) [inline config/env.config.js]: Probably use console.error instead

File: config/env.config.js (hunk 3 above)
Comment (Zlatkovsky) [reply config/env.config.js]: We're effectively looking at starts-with (indexOf === 0).  I think that's better, because we don't want query parameters or etc to mess up the logic.  As long as we start with a fully-qualified domain that's in our whitelist, we're ok.

File: config/env.config.js (hunk 4 above)
Comment (Zlatkovsky) [reply config/env.config.js]: Find returns the item, not the index.  So just the null (or undefined) is what we need to check for.
I thought about doing `!!matchingLocation`, but I think it's actually harder to understand.  I never knew about it until I saw Bhargav use it a year ago.  Is there a reason that `!!matchingLocation` is better than the `matchingLocation ? true : false` check that I'm doing?  I could be convinced either way

File: config/env.config.js (hunk 1 above)
Comment (Zlatkovsky) [reply config/env.config.js]: Fair enough.  Really, it should be the `getParameterByName` that does it.  I'll make a change.

File: config/env.config.js (hunk 4 above)
Comment (Zlatkovsky) [reply config/env.config.js]: Better yet, using Array.prototype.some:  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some
